docs/source/performance.rst

```diff
@@ -0,0 +1,41 @@
+.. _pika: https://pika.readthedocs.io/en/stable/
+.. _aio-pika: https://pika.readthedocs.io/en/stable/
+.. _msgpack: https://github.com/msgpack/msgpack-python
+
+
+Performance
+===========
+
+KiwiPy is designed primarily with ease-of-use and robustness in mind with less of a focus on performance.
+Never the less, it is expected that depending on the configuration and network conditions kiwPy should be able to handle hundreds to thousands of messages per second.
+
+The chart below shows benchmarks comparing kiwiPy to `pika`_ using the following configuration:
+
+* Sending 100 task messages consisting of 32 to 2048 bytes in size (default 1024)
+* Messages are sent and _then_ received (i.e. serialised) to be able to show split timings
+* Everything running locally (server and client) on Dell XPS 15 (Core i9-9980H @ 2.3 x 16, 1TB SSD HDD), Ubuntu 20.04
+* Using `msgpack`_ for encoding and decoding of messages
+* Mandatory routing enabled meaning that the code will wait for messages to be routed by the broker (or raise an exception)
+* Full code can be found in ``test/rmq/bench/test_benchmarks.py``
+
+Tests are split into ``send``, ``get`` and ``send_get`` to show a breakdown of the full process of sending and receiving a message.  The ``send_get`` test is carried out for different message size (in bytes) as indicated in square brackets.
+
+
+.. image:: /_static/bench.svg
+  :width: 640
+  :alt: Performance comparison of kiwiPy vs pika
+
+
+As we can see, kiwiPy is faster in getting tasks (``test_kiwi_get``: 21.7 ms vs ``test_pika__get``: 50.6 ms) but significantly slower in sending them out (``test_kiwi_send``: 237.7 ms vs ``test_pika_send``: 31.5 ms) and this results in overall (send/get) times that are roughly 3.5x (or 200ms) slower than that of pika.
+This may be due to the fact that kiwiPy is running on a separate thread from the event loop that is doing the asynchronous communication and these are synchronised each time a message is sent.
```

Comment:
with this sentence's construction it is not a 100% clear who "that is doing the" is referring to. It should be the `thread` but since it follows right after the mention of the event loop, people might get confused

Reply:
Yes, good point.  I've expanded on this a bit as I think it's useful to understand what's going on under the hood.